web/Controllers/PropertyController.cs

```diff
@@ -179,9 +176,24 @@ public async Task<IActionResult> ListPropertyImages(int propertyId)
         {
             return StatusCode(403);
         }
-        ImageListViewModel imageList = await _azureStorage.ListFiles("property", propertyId);
+        var fileNames = await _azureStorage.ListFileNames("property", propertyId);
+        var imageFiles = GetFilesFromFileNames(fileNames, propertyId);
+
+        var imageList = new ImageListViewModel()
+        {
+            PropertyId = propertyId,
+            FileList = imageFiles
+        };
+        
         return View(imageList);
     }
+
+    public List<ImageFileUrlModel> GetFilesFromFileNames(List<string> fileNames, int propertyId)
+    {
+        var imageFiles = fileNames.Select(fileName => new ImageFileUrlModel(fileName, Url.Action("GetImage", new { propertyId, fileName })!)).ToList();
+
+        return imageFiles;

```

Comment:
I think keep the url variable 😄 

```suggestion
        return fileNames.Select(fileName =>
            {
                var url = Url.Action("GetImage", new { propertyId, fileName })!;
                return new ImageFileUrlModel(fileName, url);
            })
            .ToList();
```